pyproject.toml

```
@@ -375,6 +375,12 @@ lint.ignore = [  # NOTE: non-permanent exclusions should be added to `.ruff.toml
     # flake8-todos (TD)
     "TD002",  # Missing author in TODO
 
+    # flake8-return (RET)
+    # RET505 sometimes can help find places where refactoring is very helpful,
+    # but enforcing it systematically would create undesirable churn
```

I don't like how the current comment is implying (by omission) that RET506 is not a good rule. I suspect (but I haven't proved it with examples) that there are _many_ places where explicit guard clauses would improve readability. This pull request needs to be rebased anyways to allow the documentation build to succeed.
```suggestion
    # RET can sometimes help find places where refactoring is very helpful,
    # but enforcing it everywhere might create undesirable churn
```